cypress/support/commands.js

```diff
@@ -360,6 +177,111 @@ Cypress.Commands.add('findSubmissionAsEditor', (username, password, familyName,
+	const api = new Api(Cypress.env('baseUrl') + '/index.php/publicknowledge/api/v1');
+	const hasKeywords = typeof data.keywords !== 'undefined' && data.keywords.length;
+
+	return cy.request({
+			url: api.submissions(),
+			method: 'POST',
+			headers: {
+				'X-Csrf-Token': csrfToken
+			},
+			body: {
+				sectionId: data.sectionId
+			}
+		})
+		.then(xhr => {
+			expect(xhr.status).to.eq(200);
+		})
+		.then(xhr => {
+			data.id = xhr.body.id;
+			data.currentPublicationId = xhr.body.currentPublicationId;
+			let body = {
+				title: {en_US: data.title},
+				abstract: {en_US: data.abstract},
+			};
+			if (hasKeywords) {
+				body.keywords = {en_US: data.keywords}
+			}
+			return cy.request({
+					url: xhr.body.publications[0]._href,
+					method: 'PUT',
+					headers: {
+						'X-Csrf-Token': csrfToken
+					},
+					body: body
+				})
+				.then(xhr => {
+					expect(xhr.status).to.eq(200);
+					expect(xhr.body.title.en_US).to.eq(data.title);
+					expect(xhr.body.abstract.en_US).to.eq(data.abstract);
+					if (hasKeywords) {
+						expect(xhr.body.keywords.en_US.length).to.eq(data.keywords.length);
+						data.keywords.forEach((keyword, i) => {
+							expect(xhr.body.keywords.en_US[i]).to.eq(keyword);
+						});
+					}
+				});
+		})
+		.then(xhr => {
+			if (typeof data.files === 'undefined' || !data.files.length) {
+				return xhr;
+			}
+			cy.visit('/index.php/publicknowledge/submission?id=' + data.id);
+
+			// Must use the UI to upload files until we upgrade Cypress
+			// to 7.4.0 or higher.
+			// @see https://github.com/cypress-io/cypress/issues/1647
+			cy.uploadSubmissionFiles(data.files);
+		})
+		.then(xhr => {
+			if (typeof data.additionalAuthors === 'undefined' || !data.additionalAuthors.length) {
+				return xhr;
+			}
+			data.additionalAuthors.forEach(author => {
+				cy.request({
+						url: api.contributors(data.id, data.currentPublicationId),
+						method: 'POST',
+						headers: {
+							'X-Csrf-Token': csrfToken
+						},
+						body: author
+					})
+					.then(xhr => {
+						expect(xhr.status).to.eq(200);
+					})
+			});
+			return cy.request({
+					url: api.publications(data.id, data.currentPublicationId),
+					method: 'GET'
+				})
+				.then(xhr => {
+					data.additionalAuthors.forEach(author => {
+						let publicationAuthor = xhr.body.authors.find(pAuthor => author.givenName.en_US === pAuthor.givenName.en_US);
+						expect(publicationAuthor.familyName.en_US).to.equal(author.familyName.en_US);
+						expect(publicationAuthor.affiliation.en_US).to.equal(author.affiliation.en_US);
+						expect(publicationAuthor.email).to.equal(author.email);
+						expect(publicationAuthor.country).to.equal(author.country);
+					});
+				});
+		});
+});
+
+Cypress.Commands.add('submitSubmissionWithApi', (id, csrfToken) => {
+	const api = new Api(Cypress.env('baseUrl') + '/index.php/publicknowledge/api/v1');
+	return cy.request({
+			url: api.submit(id),
+			method: 'PUT',
```

Comment:
The OPS API does not appear to support a `PUT` to this endpoint. I get:
![image](https://user-images.githubusercontent.com/200411/202316622-807388e2-ad79-43a4-af8d-00c25c3604e1.png)

Reply:
Never mind -- this seems to have been a local problem. Not sure why but it's gone away now.

Reply:
Looks like the `baseUrl` bug. I see `/undefined/` in the request URL. :+1: